[Bug]: since 1.0 rc6, linting externals

### What happened?

CI failure on https://github.com/aspect-build/aspect-workflows-template/pull/86 looks like I broke something here.

```

Lint results for //:gazelle_python_manifest.update:

external/rules_python_gazelle_plugin~/manifest/copy_to_source.py:9:8: F401 [*] `stat` imported but unused
   |
 7 | import os
 8 | import shutil
 9 | import stat
   |        ^^^^ F401
```

### Version

1.0.0-rc6 and 1.0.0-rc7

### How to reproduce

_No response_

### Any other information?

_No response_

--- a/lint/private/lint_aspect.bzl
+++ b/lint/private/lint_aspect.bzl
@@ -124,7 +124,7 @@ def filter_srcs(rule):
     if "lint-genfiles" in rule.attr.tags:
         return rule.files.srcs
     else:
-        return [s for s in rule.files.srcs if s.is_source]
+        return [s for s in rule.files.srcs if s.is_source and s.owner.workspace_name == ""]
 
 def noop_lint_action(ctx, outputs):
     """Action that creates expected outputs when no files are provided to a lint action.

